bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/AbstractComponent.java

```diff
@@ -74,20 +80,46 @@ public AbstractComponent(CFactory.ComponentConfiguration componentConfiguration,
 
         this.haID = componentConfiguration.getHaID();
 
-        String groupId = channelConfiguration.unique_id;
-        if (groupId == null || StringUtils.isBlank(groupId)) {
-            groupId = this.haID.getFallbackGroupId();
-        }
-        groupId = URLEncoder.encode(groupId).replace(".", "%DOT");
+        String groupId = this.haID.getGroupId(channelConfiguration.unique_id);
 
         this.channelGroupTypeUID = new ChannelGroupTypeUID(MqttBindingConstants.BINDING_ID, groupId);
         this.channelGroupUID = new ChannelGroupUID(componentConfiguration.getThingUID(), groupId);
+
+        this.configSeen = false;
+
+        if (StringUtils.isNotBlank(this.channelConfiguration.availability_topic)) {
+            OnOffValue value = new OnOffValue(this.channelConfiguration.payload_available,
+                    this.channelConfiguration.payload_not_available);
+
+            availablityChannel = buildChannel(HomeAssistantThingHandler.AVAILABILITY_CHANNEL, value,
+                    channelConfiguration.name + " availability").listener(componentConfiguration.getUpdateListener())//
+                            .stateTopic(channelConfiguration.availability_topic)//
+                            .build(false);
+        }
     }
 
     protected CChannel.Builder buildChannel(String channelID, Value valueState, String label) {
         return new CChannel.Builder(this, componentConfiguration, channelID, valueState, label);
     }
 
+    public void setConfigSeen() {
+        this.configSeen = true;
+    }
+
+    private @Nullable OnOffType getAvailability() {
+        @Nullable
```

Comment:
Why do you annotate here? I would guess it can be removed.

Reply:
State.as(Class<T>) is nullable as it can return <code>null</code>, if the state is not of the desired type

Reply:
Local variables are always considered as nullable prior to a null-check.